### PR TITLE
Feat/extend file session

### PR DESCRIPTION
**Pull request by prmukherj**

### CODE REVIEW

File: doc/source/api/general/data_reader.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+========
+
+The DataFile class provides a reader for Fluent data files.
+
+Sample usage
+------------
+
+You can use the DataFile class by importing it and passing a data file path.
+For some case file dependent features we need to pass the case file handle from
+the CaseFile class as well as shown in the code snippet.
```
Comment (hpohekar) [inline doc/source/api/general/data_reader.rst:13]: the CaseFile class as shown in the following code snippet.

File: doc/source/api/general/file_session.rst [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+field data from a case and data file.
+
+Sample usage
+------------
+
+You can use the FileSession by passing the case and data file information by using
+the CaseReader (:ref:`ref_case_reader`) and DataReader (:ref:`ref_data_reader`).
+This example shows adding case and data file information to FileSession. Then,
+field info is extracted followed by adding a transaction request and extracting
+field data. One ca either run a single or multi-phase case. In that respect, the example
```
Comment (hpohekar) [inline doc/source/api/general/file_session.rst:16]: One can either

**Review by hpohekar — APPROVED** [review]

File: doc/source/api/general/data_reader.rst (hunk 1 above)
Comment (prmukherj) [reply doc/source/api/general/data_reader.rst:13]: Thank you

**Review by ajain-work — APPROVED** [review]

File: doc/source/api/general/case_reader.rst [hunk 3]
```diff
@@ -50,6 +50,35 @@ Along with basic functionality, the CaseFile class provides many additional feat
...
+  repository and extract and use mesh data:
+
+  .. code-block:: python
+
+      >>> from ansys.fluent.core import examples
+      >>> from ansys.fluent.core.filereader.casereader import CaseReader
+
+      >>> case_filepath = examples.download_file("elbow1.cas.h5", "pyfluent/file_session")
+      >>> reader = CaseReader(case_filepath=case_filepath)
+      >>> reader._mesh.get_surface_ids()
```
Comment (mkundu1) [inline doc/source/api/general/case_reader.rst:65]: For a public API, change to either `reader.mesh.get_surface_ids()` or `reader.get_surface_ids()`

**Review by mkundu1 — APPROVED** [review]

File: doc/source/api/general/case_reader.rst (hunk 3 above)
Comment (prmukherj) [reply doc/source/api/general/case_reader.rst]: Done, Thank you. 'get_mesh()' was already available and has been used.

**Pull request merged**